server/server.go

```diff
@@ -514,6 +515,64 @@ func (s *server) recvLoop(stream rpcpb.ControlService_StreamStatusServer) error
 	}
 }
 
+func (s *server) AddNode(ctx context.Context, req *rpcpb.AddNodeRequest) (*rpcpb.AddNodeResponse, error) {
+	zap.L().Debug("received add node request", zap.String("name", req.Name))
+
+	if info := s.getClusterInfo(); info == nil {
+		return nil, ErrNotBootstrapped
+	}
+
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	nodeName := req.Name
```

Comment:
Can we error if the name is in conflict (e.g., `s.network.nodeInfos`)?

Can we error if the specified binary path does not exist?

Reply:
I tested this and it fails anyway later when running `s.network.nw.AddNode`.
However it makes sense to check and fail early, so will change accordingly.